File: tests/c-tests/write_emsgsize.c
// SPDX-License-Identifier: Apache-2.0

#include "libc.h"

int main(void) {
    char msg[128 * 1024];

    if (!is_enarx()) {
        return 0;
    }

    return write(STDOUT_FILENO, msg, sizeof(msg)) != -EMSGSIZE;
}

File: tests/crates/enarx_syscall_tests/src/bin/write_emsgsize.rs
// SPDX-License-Identifier: Apache-2.0

#![cfg_attr(target_os = "none", no_std)]
#![cfg_attr(target_os = "none", no_main)]

enarx_syscall_tests::startup!();

use enarx_syscall_tests::*;

fn main() -> Result<()> {
    let out = [b'A'; 128 * 1024];
    write(libc::STDOUT_FILENO, out.as_ptr(), out.len())?;
    Ok(())
}
